pkg/phlaredb/block_querier.go

```diff
@@ -983,17 +986,29 @@ func (b *singleBlockQuerier) SelectMatchingProfiles(ctx context.Context, params
 			lbls = make(phlaremodel.Labels, 0, 6)
 		}
 	}
-	pIt := query.NewJoinIterator(
-		0,
-		[]query.Iterator{
+
+	var (
+		buf       [][]parquet.Value
+		joinIters []query.Iterator
+	)
+
+	if b.meta.Version >= 2 {
```

Comment:
I think I want to include total sample in v2 WDYT ?

Reply:
I would really throw around with versions and make that version+1, also the sorted should be a version+1